flocker/node/agents/testtools/_blockdevice.py

```diff
@@ -869,3 +881,237 @@ def setUp(self):
             self.dataset_size = dataset_size
 
     return Tests
+
+
+class InvalidConfig(Exception):
+    """
+    The cloud configuration could not be found or is not compatible with the
+    running environment.
+    """
+
+
+# XXX Remember to comment on or close FLOC-2584.
+
+
+def get_blockdeviceapi():
+    """
+    Validate and load cloud provider's yml config file.
+    Default to ``~/acceptance.yml`` in the current user home directory, since
+    that's where buildbot puts its acceptance test credentials file.
+    """
+    config = get_blockdevice_config()
+    backend, api_args = backend_and_api_args_from_configuration(config)
+    api = get_api(
+        backend=backend,
+        api_args=api_args,
+        reactor=reactor,
+        cluster_id=make_cluster_id(
+            # XXX I assume that this function will only be used from functional
+            # tests.  Acceptance tests will load the API via the AgentService.
+            test_type=TestTypes.FUNCTIONAL,
+            # XXX This should match the backend, but Providers are
```

Comment:
I removed it and  completely removed the Providers constants and some unused helper functions. I *think* that's the right thing to do. At least I couldn't think of a good reason to encode "OPENSTACK", "GCE" etc into the cluster_id....that should be obvious from the cloud where we find the lingering cluster, right? (7d2ea42efa9942378f340ce76c7f7b3ff8de85b6)